external-crates/move/crates/move-compiler/tests/move_check_testsuite.rs

```diff
@@ -52,177 +66,236 @@ fn default_testing_addresses(flavor: Flavor) -> BTreeMap<String, NumericalAddres
+//         );
+//         run_test(
+//             path,
+//             Path::new(&unused_exp_path),
+//             Path::new(&unused_out_path),
+//             Flags::testing(),
+//             config.clone(),
+//             lint,
+//         )?;
+//     }
+
+//     // A cross-module unused case that should run without unused warnings suppression
+//     if path.with_extension(IDE_EXT).exists() {
+//         let ide_exp_path = format!(
+//             "{}.{IDE_EXT}.{EXP_EXT}",
+//             path.with_extension("").to_string_lossy(),
+//         );
+//         let ide_out_path = format!(
+//             "{}.{IDE_EXT}.{OUT_EXT}",
+//             path.with_extension("").to_string_lossy(),
+//         );
+//         run_test(
+//             path,
+//             Path::new(&ide_exp_path),
+//             Path::new(&ide_out_path),
+//             Flags::testing().set_ide_test_mode(true).set_ide_mode(true),
+//             config.clone(),
+//             lint,
+//         )?;
+//     }
+
+//     let exp_path = path.with_extension(EXP_EXT);
+//     let out_path = path.with_extension(OUT_EXT);
+
+//     config
+//         .warning_filter
+//         .union(&WarningFiltersBuilder::unused_warnings_filter_for_test());
+//     run_test(path, &exp_path, &out_path, Flags::empty(), config, lint)?;
+//     Ok(())
+// }
+
+fn test_config(path: &Path) -> (Option<&'static str>, TestInfo, PackageConfig, Flags) {
+    let path_extension = path.extension().unwrap();
     let path_contains = |s| path.components().any(|c| c.as_os_str() == s);
     let lint = path_contains(LINTER_DIR);
     let flavor = if path_contains(SUI_MODE_DIR) {
         Flavor::Sui
     } else {
         Flavor::default()
     };
-    let edition = if path_contains(MOVE_2024_DIR) {
+    let migration_mode = path_extension == MIGRATION_EXT;
+    let move_2024_mode = path_contains(MOVE_2024_DIR);
+    let dev_mode = path_contains(DEV_DIR);
+    assert!(
+        [move_2024_mode, dev_mode,]
+            .into_iter()
+            .filter(|x| *x)
+            .count()
+            <= 1,
+        "A test can have at most directory based edition"
+    );
+    let edition = if migration_mode {
+        // migration mode overrides the edition
+        Edition::E2024_MIGRATION
+    } else if move_2024_mode {
         Edition::E2024_ALPHA
-    } else if path_contains(DEV_DIR) {
+    } else if dev_mode {
         Edition::DEVELOPMENT
     } else {
         Edition::LEGACY
     };
-    let config = PackageConfig {
+    // test kind
+    let test_kind = match () {
+        _ if path_extension == MOVE_EXTENSION => None,
+        _ if path_extension == TEST_EXT => Some(TEST_EXT),
+        _ if path_extension == UNUSED_EXT => Some(UNUSED_EXT),
+        _ if path_extension == MIGRATION_EXT => Some(MIGRATION_EXT),
+        _ if path_extension == IDE_EXT => Some(IDE_EXT),
+        _ => panic!("Unknown extension: {}", path_extension.to_string_lossy()),
```

Comment:
What about making this a match? Or is `path_extension` a path of some kinda and `match` doesn't like that? i.e., what about something like

```rust
let test_kind = match path_extension {
   MOVE_EXTENSION = NONE,
   x @ (TEST_EXT | UNUSED_EXT | MIGRATION_EXT | IDE_EXT) => Some(x),
   _ => panic!(...),
};
```